dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/repository/GitHub.java

```diff
@@ -24,4 +27,32 @@ public static boolean shouldNotProcessDockerfilesInRepo(Multimap<String, String>
         }
         return false;
     }
+
+    /**
+     * Check to see if latest sha from parent is in the tree of the fork. If not, something has happened in GitHub to
+     * make the commit graph inconsistent. We'll just call it "stale" for brevity. GitHub Enterprise administrators
+     * can fix this by running "Rebuild Network Graph" in the administrative user interface. Deleting the fork can work,
+     * but that results in long delays since the API thinks the fork is still there after deletion for some long period
+     * of time.
+     * If we administrators don't rebuild the network graph, we get a "422 Reference update failed" if we try and
+     * create a new branch. https://docs.github.com/enterprise/2.22/rest/reference/git#create-a-reference
+     * Updating the default branch of the fork to the sha results in "422 Object does not exist"
+     * https://docs.github.com/enterprise/2.22/rest/reference/git#update-a-reference
+     *
+     * @param parent the parent
+     * @param fork the fork associated with parent
+     * @return the fork is stale for some reason
+     * @throws IOException general issues trying to get information about the parent's branch will throw an IOException
+     */
+    public static boolean isForkStale(GHRepository parent, GHRepository fork) throws IOException {
+        String sha1 = parent.getBranch(parent.getDefaultBranch()).getSHA1();
+        try {
+            fork.getTree(sha1);
+        } catch (GHFileNotFoundException e) {
+            log.info("fork {} is stale... A GitHub administrator will need to rebuild the network graph.",
```

Comment:
Is there a procedure for doing that?

Reply:
Yeah, via admin user interface only unfortunately.